File: sdk-java/src/main/java/com/tollara/client/model/UsageReportResponse.java
package com.tollara.client.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

/**
 * Response model from the usage service after reporting usage (reportSchemaVersion 2).
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class UsageReportResponse {
    private int reportSchemaVersion;
    private String status;
    private String warning;
    private String userId;
    private String serviceId;
    private String billingModelType;
    private String measurementType;
    private String unitLabel;
    private UsageBreakdown breakdown;
}

File: sdk-php/src/UsageReportResponse.php
<?php

declare(strict_types=1);

namespace Tollara\ServiceSdk;

final class UsageReportResponse
{
    public function __construct(
        public int $reportSchemaVersion,
        public ?string $status,
        public ?string $warning,
        public ?string $userId,
        public ?string $serviceId,
        public ?string $billingModelType,
        public ?string $measurementType,
        public ?string $unitLabel,
        public ?UsageBreakdown $breakdown,
    ) {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        $breakdown = null;
        if (isset($data['breakdown']) && is_array($data['breakdown'])) {
            $breakdown = UsageBreakdown::fromArray($data['breakdown']);
        }
        return new self(
            reportSchemaVersion: (int) ($data['reportSchemaVersion'] ?? 0),
            status: isset($data['status']) ? (string) $data['status'] : null,
            warning: isset($data['warning']) ? (string) $data['warning'] : null,
            userId: isset($data['userId']) ? (string) $data['userId'] : null,
            serviceId: isset($data['serviceId']) ? (string) $data['serviceId'] : null,
            billingModelType: isset($data['billingModelType']) ? (string) $data['billingModelType'] : null,
            measurementType: isset($data['measurementType']) ? (string) $data['measurementType'] : null,
            unitLabel: isset($data['unitLabel']) ? (string) $data['unitLabel'] : null,
            breakdown: $breakdown,
        );
    }
}
